Mathlib/Data/Matrix/Rank.lean

```diff
@@ -198,6 +200,16 @@ proof that is a simple consequence of `Matrix.rank_transpose_mul_self` and
 be replaced with a proof that uses Gaussian reduction or argues via linear combinations.
 -/
 
+section Field
+
+variable  [Field R]
+/-- The rank of a diagnonal matrix  is the count of non-zero elements on its main diagonal -/
+theorem rank_diagonal [DecidableEq m] [DecidableEq R] (w : m → R) :
+    ((Matrix.diagonal w).rank) = (Fintype.card {i // (w i) ≠ 0}) := by
+  rw [Matrix.rank, ← Matrix.toLin'_apply', FiniteDimensional.finrank, ← LinearMap.rank,
+    LinearMap.rank_diagonal, Cardinal.toNat_cast]
+
+end Field
```

Comment:
This is in the wrong place; the section above says these are lemmas about tranpose and conj_transpose. Can you move it above that comment?